lib/repl.js

```diff
@@ -534,9 +544,50 @@ function REPLServer(prompt,
   });
 
   self.on('SIGCONT', function() {
-    self.displayPrompt(true);
+    if (self.editorMode) {
+      self.outputStream.write(`${self._initialPrompt}.editor\n`);
+      self.outputStream.write(
+        '// Entering editor mode (^D to finish, ^C to cancel)\n');
+      self.outputStream.write(`${self.bufferedCommand}\n`);
+      self.prompt(true);
+    } else {
+      self.displayPrompt(true);
+    }
   });
 
+  // Wrap readline tty to enable editor mode
+  const ttyWrite = self._ttyWrite.bind(self);
+  self._ttyWrite = (d, key) => {
+    if (!self.editorMode || !self.terminal) {
+      ttyWrite(d, key);
```

Comment:
@thefourtheye 

`self._ttyWrite` and `ttyWrite` are different functions. `ttyWrite` has access to original `self._ttyWrite` and is bounded by `self` reference.

`object.method` is a nice shortcut for languages which supports single dispatch, so it is same as
`method(object)`.

`ttyWrite => global.ttyWrite => ttyWrite(global)`